components/metadata-service/internal/metadata/servicedefservice.go

```diff
@@ -54,9 +65,17 @@ func NewServiceDefinitionService(uuidGenerator uuid.Generator, serviceAPIService
 
 // Create adds new ServiceDefinition. Based on ServiceDefinition a new service is added to RemoteEnvironment.
 func (sds *serviceDefinitionService) Create(remoteEnvironment string, serviceDef *ServiceDefinition) (string, apperrors.AppError) {
+	if serviceDef.Identifier != "" {
+		err := sds.ensureUniqueIdentifier(serviceDef.Identifier, remoteEnvironment)
+		if err != nil {
+			return "", err
+		}
+	}
+
 	id := sds.uuidGenerator.NewUUID()
 
-	service := initService(serviceDef, id)
+	service := initService(serviceDef, id, remoteEnvironment)
+	service.Identifier = serviceDef.Identifier
```

Comment:
The update function removes `identifier` whether it is specified or not. Also the payload returned by update function contains the identifier sent in PUT request despite it is not being updated.